macosfrontend/macosfrontend.swift

```diff
@@ -38,7 +38,10 @@ public func commitAndSetPreeditSync(
   // Without client preedit, Backspace bypasses IM in Terminal, every key
   // is both processed by IM and passed to client in iTerm, so we force a
   // dummy client preedit here.
-  if preedit.isEmpty && dummyPreedit {
+  // Some apps also need it to get accurate cursor position to place candidate window.
+  // But when user selects some text using Shift, we don't want the dummy
```

Comment:
The wording can be improved, e.g. when there is selected text.

Reply:
Sure, though I'd preserve the Shift-click as an example as it's really annoying and I don't want it be broken again.